middleman_protocol/middleman_protocol/helpers.py

```diff
@@ -0,0 +1,18 @@
+from asyncio import StreamReader
+
+from middleman_protocol.constants import ErrorCode
+from middleman_protocol.constants import FRAME_SEPARATOR
+from middleman_protocol.constants import MIDDLEMAN_EXCEPTION_TO_ERROR_CODE_MAP
+from middleman_protocol.exceptions import MiddlemanProtocolError
+from middleman_protocol.stream import escape_decode_raw_message
+
+
+async def handle_frame_receive_async(reader: StreamReader) -> bytes:
+    raw_data = await reader.readuntil(FRAME_SEPARATOR)
```

Comment:
Please add checking if `MAXIMUM_FRAME_LENGTH` has been exceeded.

Reply:
`StreamReader` already have that kind of check inside it's class. `_DEFAULT_LIMIT = 65536`. In case of exceeding the limit, we will get `LimitOverrunError` (https://docs.python.org/3/library/asyncio-stream.html#asyncio.StreamReader.readuntil). Maybe we should add handler for this and other errors, where `handle_frame_receive_async()` is being called, since it's already inside `try/except` block?

Reply:
Ok, then I think you should use the `MAXIMUM_FRAME_LENGTH` constant when creating `StreamReader`. This way both implementations (sync and async) will have the same feature. 

In sync version, if the limit is exceeded, socket continues to receive data until separator, but doesn't yield that data, and after separator is received, it starts clears data and start over. I am now thinking if it shouldn't return `None`. @dybi @cameel what do you guys think ?

Reply:
Okey, I've added `limit=MAXIMUM_FRAME_LENGTH` where we run `asyncio.start_server()`. It works like the docs say.